test/ret_test.exs

```diff
@@ -0,0 +1,681 @@
+
+      assert :ok === Ret.delete_account(account_to_delete.account_id, current_user)
+      refute repo_reload(favorite1)
+      refute repo_reload(favorite2)
+    end
+
+    test "deletes API credentials", %{account_to_delete: account_to_delete, current_user: current_user} do
+      Api.TokenUtils.gen_token_for_account(account_to_delete)
+      credentials = Repo.get_by(Api.Credentials, account_id: account_to_delete.account_id)
+
+      assert :ok === Ret.delete_account(account_to_delete.account_id, current_user)
+      refute repo_reload(credentials)
+    end
+
+    test "deletes assets", %{account_to_delete: account_to_delete, current_user: current_user} do
+      asset = create_asset(account_to_delete)
+
+      %Asset{} = repo_reload(asset)
+
+      assert :ok === Ret.delete_account(account_to_delete.account_id, current_user)
+      refute repo_reload(asset)
+    end
+
+    test "deletes asset owned files", %{account_to_delete: account_to_delete, current_user: current_user} do
+      asset = create_asset(account_to_delete)
+      owned_files = owned_files(asset, [:asset_owned_file, :thumbnail_owned_file])
+
+      for owned_file <- owned_files do
+        %OwnedFile{} = repo_reload(owned_file)
+        true = file_on_disk?(owned_file)
+      end
+
+      assert :ok === Ret.delete_account(account_to_delete.account_id, current_user)
+
+      for owned_file <- owned_files do
+        refute repo_reload(owned_file)
+        refute file_on_disk?(owned_file)
+      end
+    end
+
+    test "deletes avatars", %{account_to_delete: account_to_delete, current_user: current_user} do
+      avatar = create_avatar(account_to_delete)
+
+      %Avatar{} = repo_reload(avatar)
+
+      assert :ok === Ret.delete_account(account_to_delete.account_id, current_user)
+      refute repo_reload(avatar)
+    end
+
+    test "deletes avatar owned files", %{account_to_delete: account_to_delete, current_user: current_user} do
+      avatar = create_avatar(account_to_delete)
+      owned_files = avatar_owned_files(avatar)
+
+      for owned_file <- owned_files do
+        %OwnedFile{} = repo_reload(owned_file)
+        true = file_on_disk?(owned_file)
+      end
+
+      assert :ok === Ret.delete_account(account_to_delete.account_id, current_user)
+
+      for owned_file <- owned_files do
+        refute repo_reload(owned_file)
+        refute file_on_disk?(owned_file)
+      end
+    end
+
+    test "reassigns parent avatars to current user", %{account_to_delete: account_to_delete, current_user: current_user} do
+      avatar = create_avatar(account_to_delete)
+      account_to_delete_id = account_to_delete.account_id
+      create_child_avatar(avatar)
+
+      %Avatar{account_id: ^account_to_delete_id} = repo_reload(avatar)
+
+      assert :ok === Ret.delete_account(account_to_delete.account_id, current_user)
+      assert avatar = repo_reload(avatar)
+      assert current_user.account_id === avatar.account_id
```

Comment:
The idea here is to prevent breaking any avatars down the remix chain from the deleted account, by moving ownership of that over to the admin account that is currently doing the deleting? I guess that makes sense but not sure how it works in terms of compliance with deleting all of a users data... I guess remixing was a specific opt in that allowed us a CC_BY license to the content. https://github.com/mozilla/hubs/blob/master/REMIXING.md

Reply:
Correct. As soon as content is made promotable, it becomes public data.

By analogy, if someone were to remove their repo from GitHub, you would be shocked if your fork of that repo were to disappear along with the original! 😱